blog/en/blog/2022/10/27/ten-use-cases-api-gateway.md

```diff
@@ -0,0 +1,180 @@
+
+![API geo-routing](https://static.apiseven.com/2022/10/25/6357afebdc9c0.png)
+
+As you can see in the preceding diagram, it uses a DNS, traffic management service, and an API Gateway to resolve each subdomain against the region’s load balancer and passes the client request further down to the closest API Gateway.
+
+## API aggregator
+
+This technique performs operations (for example, queries) against multiple services and returns the result to the client service with a single `HTTP request/response` call. Instead of having a client application make several calls to multiple APIs, an API aggregator uses an API Gateway to do this on behalf of the consumer on the server side.
+
+For example, consider a mobile app that makes multiple calls to different APIs to show the data for a single screen. In this case, it increases complexity in the client-side code, over-utilization of network resources, and even poor user experience as the application is more exposed to latency issues. API Gateway can accept as input all information required, does request authentication and validation, understands all data structures from all APIs that it interacts with, and is capable of transforming the response payloads so they can be sent back to the mobile app as a uniform payload needed for the consumer.
+
+![API geo-routing](https://static.apiseven.com/2022/10/25/6357a5017297b.png)
+
+## API Centralized Authentication
+
+In this design, an API Gateway acts as a **centralized authentication gateway**. As an authenticator, API Gateway looks for access credentials in the `HTTP header` - for example, a bearer token, and implements business logic that validates those credentials with an IDP, identity provider such as [Okta](https://www.okta.com/), [Cognito](https://aws.amazon.com/cognito/), [Azure Active Directory](https://azure.microsoft.com/en-us/services/active-directory/) or [Ory Hydra](https://www.ory.sh/hydra/)) typically using the [OpenID Connect](https://openid.net/connect/) (OIDC) is an authentication protocol based on the [OAuth 2.0](https://oauth.net/2/) and checking whether they have access or not.
+
+![Header based basic authentication](https://static.apiseven.com/2022/10/25/6357b6a55569a.png)
+
+Centralized authentication with the API Gateway can solve many problems and have some benefits as it completely offloads user management from an application and it improves performance by responding quickly to authentication requests received from client applications.
+
+![identityprovider](https://static.apiseven.com/2022/10/25/6357a75d9da89.jpg)
```

Comment:
Nitpick: why do you use a sketchy diagram representation while the rest isn't?

Reply:
Because, sketchy diagrams I draw new, the rest from old content:) It consumes a lot of time.